qemu/tests/live_backup_base.py

```diff
@@ -0,0 +1,214 @@
+                               "format": backup_format,
+                               "sync": sync,
+                               "mode": mode,
+                               "speed": speed}
+                self.transaction_add(args_list, "drive-backup", backup_args)
+                error_context.context("Create bitmap and drive-backup with transaction "
+                                      "for %s" % drive_name, logging.info)
+                self.vm.monitor.transaction(args_list)
+                if not self.get_status():
+                    raise exceptions.TestFail("%s backup job not found" % sync)
+                return None
+
+            error_context.context("Create bitmap for %s" % drive_name, logging.info)
+            self.vm.monitor.operate_dirty_bitmap("add", drive_name, bitmap_name, granularity)
+        if not backup_image_name:
+            raise exceptions.TestError("No backup target provided.")
+        error_context.context("Create %s backup for %s" % (sync, drive_name), logging.info)
+        self.vm.monitor.drive_backup(drive_name, backup_image_name, backup_format,
+                                     sync, speed, mode, bitmap_name)
+        if not self.get_status():
+            raise exceptions.TestFail("%s backup job not found" % sync)
+        utils_memory.drop_caches()
+
+    def transaction_add(self, args_list, type, data):
+        """
+        Generate a args list for a transaction.
+        :param args_list: the arg_list param that will be modified
+        :param type: type of the command
+        :param data: data of the command
+        """
+        args = {"type": type, "data": data}
+        args_list.append(args)
+
+    def backup_check(self, compare_image=None):
+        """
+        Check and compare the backup images with qemu-img
+        :param compare_image: the image that need to be compared
+        """
+        data_dir = self.data_dir
+        for image in self.image_chain:
+            params = self.params.object_params(image)
+            qemu_image = qemu_storage.QemuImg(params, data_dir, image)
+            qemu_image.check_image(params, data_dir)
+        params = self.params.object_params(self.source_image)
+        if compare_image:
+            compare_image = utils_misc.get_path(data_dir, compare_image)
+            source_image = qemu_storage.storage.get_image_filename(params, data_dir)
+            qemu_image.compare_images(compare_image, source_image)
+
+    def reopen(self):
+        """
+        Closing the vm and reboot it with the backup image.
+        """
+        vm = self.vm
+        vm.destroy()
+        image_chain = self.image_chain
+        image_name = self.params.get("image_name_%s" %
+                                     image_chain[-1])
+        self.params["image_name_%s" % self.source_image] = image_name
+        vm.create(params=self.params)
+        self.vm = vm
+        session = self.get_session()
+        session.close()
```

Comment:
it looks no need to do it here, all session will closed in the end of test. but it's harmless. And, I think self.vm.verify_alive() looks more reasonable here.